rocketpy/Motor.py

```diff
@@ -0,0 +1,1937 @@
+        general shape of the curve. May translate the curve so that
+        thrust starts at time equals 0, without any delays.
+
+        Parameters
+        ----------
+        burnTime : float
+            New desired burn out time in seconds.
+        totalImpulse : float
+            New desired total impulse.
+        oldTotalImpulse : float, optional
+            Specify the total impulse of the given thrust curve,
+            overriding the value calculated by numerical integration.
+            If left as None, the value calculated by numerical
+            integration will be used in order to reshape the curve.
+        startAtZero: bool, optional
+            If True, trims the initial thrust curve points which
+            are 0 Newtons, translating the thrust curve so that
+            thrust starts at time equals 0. If False, no translation
+            is applied.
+
+        Returns
+        -------
+        None
+        """
+        # Retrieve current thrust curve data points
+        timeArray = self.thrust.source[:, 0]
+        thrustArray = self.thrust.source[:, 1]
+        # Move start to time = 0
+        if startAtZero and timeArray[0] != 0:
+            timeArray = timeArray - timeArray[0]
+
+        # Reshape time - set burn time to burnTime
+        self.thrust.source[:, 0] = (burnTime / timeArray[-1]) * timeArray
+        self.burnOutTime = burnTime
+        self.thrust.setInterpolation(self.interpolate)
+
+        # Reshape thrust - set total impulse
+        if oldTotalImpulse is None:
+            oldTotalImpulse = self.evaluateTotalImpulse()
+        self.thrust.source[:, 1] = (totalImpulse / oldTotalImpulse) * thrustArray
+        self.thrust.setInterpolation(self.interpolate)
+
+        # Store total impulse
+        self.totalImpulse = totalImpulse
+
+        # Return reshaped curve
+        return self.thrust
+
+    def evaluateTotalImpulse(self):
+        """Calculates and returns total impulse by numerical
+        integration of the thrust curve in SI units. The value is
+        also stored in self.totalImpulse.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        self.totalImpulse : float
+            Motor total impulse in Ns.
+        """
+        # Calculate total impulse
+        self.totalImpulse = self.thrust.integral(0, self.burnOutTime)
+
+        # Return total impulse
+        return self.totalImpulse
+
+    @abstractproperty
+    def exhaustVelocity(self):
+        """Calculates and returns exhaust velocity by assuming it
+        as a constant. The formula used is total impulse/propellant
+        initial mass. The value is also stored in
+        self.exhaustVelocity.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        self.exhaustVelocity : float
+            Constant gas exhaust velocity of the motor.
+        """
+        pass
+
+    @abstractmethod
+    def evaluateMassDot(self):
```

Comment:
Should this method and similar ones be converted to abstract properties just as was done for exhaust velocity?

Reply:
Perhaps it should? Seems like a matter of preference to me. Up until now I've been defining as properties simple calculations which require 1-2 lines of code, and methods for everything else. Do you have any specific criteria we could use in mind?